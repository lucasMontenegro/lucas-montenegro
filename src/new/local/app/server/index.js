const cluster = require("cluster")
const os = require("os")
const { default: startHttp } = require("new/local/app/server/startHttp")
const { default: handleClusterExit } = require("new/local/app/server/handleClusterExit")
const { default: globals } = require("new/local/utils/globals")
exports.default = function start () {
  // Multi-process to utilize all CPU cores.
  if (globals.process.env.NODE_ENV === `production` && cluster.isMaster) {
    globals.console.log(`Node cluster master ${globals.process.pid} is running`)
    // Fork workers.
    for (let i = 0, n = os.cpus().length; i < n; i++) {
      cluster.fork()
    }
    cluster.on(`exit`, handleClusterExit)
  } else {
    startHttp()
  }
}