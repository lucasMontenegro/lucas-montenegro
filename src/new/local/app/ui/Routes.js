import React from "react"
import { BrowserRouter, Switch, Route } from "react-router-dom"
import App from "new/local/app/ui/App"
import Examples from "new/local/paperbase/Examples"
import isProduction from "new/local/utils/isProduction"
export default function Routes () {
  return (
    <BrowserRouter>
      <Switch>
        <Route component={App} />
        {isProduction() || <Route path="/examples" component={Examples} />}
      </Switch>
    </BrowserRouter>
  )
}