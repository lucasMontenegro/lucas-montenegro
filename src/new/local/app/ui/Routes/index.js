import React from "react"
import { BrowserRouter, Switch, Route } from "react-router-dom"
import isProduction from "new/local/utils/isProduction"
import App from "new/local/app/ui/App"
import Examples from "new/local/app/ui/Routes/Examples"
import OldExamples from "new/local/app/ui/Routes/OldExamples"
export default function Routes () {
  return (
    <BrowserRouter>
      {
        isProduction() ? <Route component={App} /> :
        (
          <Switch>
            <Route path="/examples" component={Examples} />
            <Route path="/old-examples" component={OldExamples} />
            <Route component={App} />
          </Switch>
        )
      }
    </BrowserRouter>
  )
}