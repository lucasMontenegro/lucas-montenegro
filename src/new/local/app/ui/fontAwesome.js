const { library } = require("@fortawesome/fontawesome-svg-core")
const { fab } = require("@fortawesome/free-brands-svg-icons")
const { fas } = require("@fortawesome/free-solid-svg-icons")
const { far } = require("@fortawesome/free-regular-svg-icons")
library.add(fab, fas, far)