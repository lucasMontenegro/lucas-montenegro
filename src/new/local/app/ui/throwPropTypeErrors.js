const { default: globals } = require("new/local/utils/globals")
const { default: isProduction } = require("new/local/utils/isProduction")
exports.default = function throwPropTypeErrors () {
  if (isProduction()) {
    return
  }
  const { error } = globals.console
  globals.console.error = function logErrorOrThrow (msg) {
    if (/invalid|failed/gi.test(msg) && /prop.?type/gi.test(msg)) {
      throw new Error(msg)
    }
    error.apply(this, arguments)
  }
}