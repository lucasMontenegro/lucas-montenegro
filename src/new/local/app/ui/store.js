import React from "react"
import { createStore, combineReducers } from "redux"
import { Provider } from "react-redux"
import Routes from "new/local/app/ui/Routes"
export const reducers = {
  core (state={}, action) {
    return state
  },
}
export default (
  <Provider store={createStore(combineReducers(reducers))}>
    <Routes />
  </Provider>
)