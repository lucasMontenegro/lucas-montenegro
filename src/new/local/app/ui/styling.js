const React = require("react")
const { default: CssBaseline } = require("@material-ui/core/CssBaseline")
const { ThemeProvider } = require("@material-ui/styles")
const { createMuiTheme } = require("@material-ui/core/styles")
require("typeface-roboto")
require("new/local/app/ui/fontAwesome")
const { default: store } = require("new/local/app/ui/store")
exports.default = (
  <ThemeProvider theme={createMuiTheme()}>
    <CssBaseline />
    {store}
  </ThemeProvider>
)