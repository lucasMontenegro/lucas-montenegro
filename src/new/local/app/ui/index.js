import ReactDOM from "react-dom"
import styling from "new/local/app/ui/styling"
import throwPropTypeErrors from "new/local/app/ui/throwPropTypeErrors"
import globals from "new/local/utils/globals"
throwPropTypeErrors()
ReactDOM.render(styling, globals.document.getElementById(`root`))