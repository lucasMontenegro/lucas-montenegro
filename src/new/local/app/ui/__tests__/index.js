jest.mock(`react-dom`, () => ({
  default: { render: jest.fn() },
}))
jest.mock(`new/local/app/ui/styling`, () => ({ default: `new/local/app/ui/styling` }))
jest.mock(`new/local/app/ui/throwPropTypeErrors`, () => ({ default: () => {} }))
jest.mock(`new/local/utils/globals`, () => ({
  default: {
    document: { getElementById: () => {} },
  },
}))
describe(`new/local/app/ui`, () => {
  it(`should run`, () => {
    require("new/local/app/ui")
  })
})