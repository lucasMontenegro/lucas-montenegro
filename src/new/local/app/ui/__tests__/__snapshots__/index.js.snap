// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new/local/app/ui should utilize the dependency APIs correctly 1`] = `
Object {
  "react-dom": "^16.10.1",
}
`;
